nemo/backends/pytorch/actions.py

```diff
@@ -526,6 +526,18 @@ def _eval(self, tensors_2_evaluate, callback, step, verbose=False):
                         tensors_list = []
                         # where we will all_gather tensor sizes
                         tensor_on_worker = registered_e_tensors[key]
+
+                        if not isinstance(tensor_on_worker, torch.Tensor):  # For string and other.
+                            if self.global_rank == 0:
+                                values_dict[key] = [tensor_on_worker] + ([None] * (world_size - 1))
+                            continue
```

Comment:
Image I want to log something with raw python strings.
For example (my case): I use this for logging audio + raw text transcript (as transcription) for W&B.

Reply:
How are you passing a raw python string inside of NeMo? Last I checked, NeMo does not support non-tensors

Reply:
It's actually does. The trick is `text_raw=NeuralType(),`.

Reply:
@okuchaiev or @tkornuta-nvidia
Can you guys comment on this change? And make a decision on whether we want to make this change? I am conflicted about accepting this.

Reply:
There is no other way if you want to log something with python objects (from my pov).

Reply:
Hey, we have discussed that with @blisc and concluded that this will work. Actually I was discussing the use-cases with @ekmb  some time ago where she wanted to send strings (in SGD). As far as I remember she was using torch.distributed, it was raising an error, it seems to me that you just also solved her issue ;)

Thanks @stasbel 👍

Reply:
Actually I think I got a use case that will not work here.

List of bounding boxes in ObjectDetection. I see that as list (batch) of list of 1D, 4-element tensors.

I think I got an idea (actually options) how to refine this, probably it should be discussed once I will be working on OD - which is phase III of CV collection development, and we have just passed stage I. So this is not going to happen soon.

That is why I still think it should be merged.

Reply:
As far as NeuralTypes are concerned "string", "float", "float16", "int" are "storage" types which we don't enforce unless the user explicitly declares it in a way similar to how we declare sampling rate.
So, technically, you should be able to declare something like List(Tensor[string]). But this is only used to enforce validity of connections. We currently do not check if module's implementation actually obeys it's types.

Speaking of this particular change - it makes sense to me, so I'm OK for including this.